### ISSUE TITLE
Windows build fails with bash in C:\Windows\System32\bash.exe

### ISSUE THREAD

**Issue by SaschaWillems** [KhronosGroup/KTX-Software#405]
The Windows CMake using MSVC 2019 om Windows 10 fails for me with a missing bash.exe with Code 9009:

```
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(240
,5): error MSB8066: Der benutzerdefinierte Build für "V:\github\KTX-Software\build\CMakeFiles\1972c4c34ae512daa92f80677
83b4cf5\version.h.rule;V:\github\KTX-Software\build\CMakeFiles\5536772b8d47a00d5d4d7b857032fd8a\ktx_version.rule;V:\git
hub\KTX-Software\CMakeLists.txt" wurde mit dem Code 9009 beendet. [V:\github\KTX-Software\build\ktx_version.vcxproj]
  Generating VkFormat-related source files
  Der Befehl "C:\Windows\System32\bash.exe" ist entweder falsch geschrieben oder
  konnte nicht gefunden werden.
```

Sorry for the german error, but that's my system language. The error states the the command "C:\Windows\System32\bash.exe" is misspelled or cannot be found.

On my system bash is installed in "C:\Windows\System32\bash.exe", afair this came via WSL.

It fails both from the command line, using the commands from the build instructions, as well as from the IDE, so it seems the build has troubles trying to launch bash from that location.

MSVC version: 16.9.4
CMake version: 3.20.2

**Comment by MarkCallow** [comment]
Look in your CMake configuration for the variable `BASH_EXECUTABLE`. Probably easiest to do with the CMake GUI. You may have to click the "Advanced" checkbox to see it. Is it set, if so what to? As long as it is set to where your bash is located, the build should work. Windows is one of our CI targets. CI runs builds on VS2015, VS2017 and VS2019.

**Comment by SaschaWillems** [comment]
```BASH_EXECUTABLE``` points to ```C:\Windows\System32\bash.exe```:

![image](https://user-images.githubusercontent.com/10283127/115954087-4bc0d600-a4ef-11eb-9433-dbafde3badb0.png)

And that bash.exe works fine when calling it manually.

**Comment by MarkCallow** [comment]
BASH_EXECUTABLE is set using `/` not `\`. The only thing I can think of is that somehow the `/`s have not been changed to `\` when the projects were generated. In my own case it is also defined using `/` but the generated project has `\`. Do

```bash
grep mkversion ktx_version.vcxproj
```

in your build folder. (Sorry about the grepping. I haven't been able to figure out how to view these custom commands through the Visual Studio GUI.) You should see some lines like

```
"c:\Windows\System32\bash.exe" -c "\"./mkversion\" \"-o version.h\" \"lib\""
```

If you see `/` then the command will fail as, IIRC, VS uses cmd.exe to run these commands.

Also check that `C:\Windows\System32\` is in the list of Executable Directories in the VC++ Directories section of the properties of the *_version projects in the VS solution.

**Comment by SaschaWillems** [comment]
Just checked this, and all requirements are met. I can't tell why this fails.

Replacing the automatically detected bash.exe from the windows folder with the one from my git folder does not longer fail the build, but now I see a few warnings about the ```version.h``` file not being generated:

```
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\1972c4c34ae512daa92f8067783b4cf5\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\lib\version.h" has not been created. This may cause incremental build to work incorrectly.	ktx_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\9c0961027b23efbf5deb0f0e7261f506\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\tools\ktx2check\tools\ktx2check\version.h" has not been created. This may cause incremental build to work incorrectly.	ktx2check_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\bd72429456776347336b9242a5d53f1a\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\tools\ktx2ktx2\tools\ktx2ktx2\version.h" has not been created. This may cause incremental build to work incorrectly.	ktx2ktx2_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\d25870cd20e80180798fcfb70a485014\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\tools\ktxsc\tools\ktxsc\version.h" has not been created. This may cause incremental build to work incorrectly.	ktxsc_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\a477664e6a7c3d0fe87595e2fcc84e28\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\tools\ktxinfo\tools\ktxinfo\version.h" has not been created. This may cause incremental build to work incorrectly.	ktxinfo_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	
Warning	MSB8065	Custom build for item "V:\github\KTX-Software\build\CMakeFiles\41b00962128edf045585b316465b307f\version.h.rule" succeeded, but specified output "v:\github\ktx-software\build\tools\toktx\tools\toktx\version.h" has not been created. This may cause incremental build to work incorrectly.	toktx_version	C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets	240	

```

Not sure if this is correct, but at least I can build the project now. So the initial issue may only affect people using WSL on Windows which automatically installs bash.

**Comment by MarkCallow** [comment]
Those warnings were fixed in commit fa29fe384b. Please pull the latest code.
